src/controllers/keys.ts

```diff
@@ -0,0 +1,40 @@
+import { Request, Response } from "express";
+import { prisma } from "../utils/connect";
+
+export const getAllKeysForContract = async (
+  req: Request,
+  res: Response,
+): Promise<void | Response> => {
+  try {
+    const { contract_id, network = "mainnet" } = req.params;
+
+    if (network !== "mainnet") {
+      return res.status(400).json({ error: "Only mainnet is supported" });
+    }
+
+    // Use Prisma ORM to get distinct key_decoded values
+    const result = await prisma.contract_data.findMany({
+      where: {
+        id: contract_id,
+      },
+      select: {
+        key_decoded: true,
+      },
+      distinct: ["key_decoded"],
```

Comment:
Following the natural order of a SQL query would make this more readable:


```suggestion
      distinct: ["key_decoded"],
      select: {
        key_decoded: true,
      },
      where: {
        id: contract_id,
      },
```